src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1

```diff
@@ -421,36 +448,43 @@ function Set-AzConnectedKubernetes {
                 $DisableAutoUpgrade = ($InputObject.ArcAgentProfileAgentAutoUpgrade -eq 'Disabled')
             }
 
-            if ((-not $PSBoundParameters.ContainsKey('WorkloadIdentityEnabled')) -and $InputObject.PSObject.Properties['WorkloadIdentityEnabled']) {
-                $WorkloadIdentityEnabled = $InputObject.WorkloadIdentityEnabled
-                $PSBoundParameters.Add('WorkloadIdentityEnabled', $WorkloadIdentityEnabled)
-            }
-
-            if ((-not $PSBoundParameters.ContainsKey('OidcIssuerProfileEnabled')) -and $InputObject.OidcIssuerProfileEnabled) {
-                $OidcIssuerProfileEnabled = $true
-                $PSBoundParameters.Add('OidcIssuerProfileEnabled', $OidcIssuerProfileEnabled)
-            }
-
-            if ((-not $PSBoundParameters.ContainsKey('OidcIssuerProfileSelfHostedIssuerUrl')) -and $InputObject.OidcIssuerProfileSelfHostedIssuerUrl) {
-                $OidcIssuerProfileEnabled = $true
-                $PSBoundParameters.Add('OidcIssuerProfileSelfHostedIssuerUrl', $InputObject.OidcIssuerProfileSelfHostedIssuerUrl)
-            }
-
-            if ((-not $PSBoundParameters.ContainsKey('Distribution')) -and $InputObject.PSObject.Properties['Distribution']) {
-                $PSBoundParameters.Add('Distribution', $InputObject.Distribution)
-            }
-
-            if ((-not $PSBoundParameters.ContainsKey('DistributionVersion')) -and $InputObject.PSObject.Properties['DistributionVersion']) {
-                $PSBoundParameters.Add('DistributionVersion', $InputObject.DistributionVersion)
-            }
-
-            if ((-not $PSBoundParameters.ContainsKey('Infrastructure')) -and $InputObject.PSObject.Properties['Infrastructure']) {
-                $PSBoundParameters.Add('Infrastructure', $InputObject.Infrastructure)
-            }
-
-            if ((-not $PSBoundParameters.ContainsKey('PrivateLinkState')) -and $InputObject.PSObject.Properties['PrivateLinkState']) {
-                $PSBoundParameters.Add('PrivateLinkState', $InputObject.PrivateLinkState)
-            }
+            Set-PSDebug 2
+
+            # Merge the fields that use a common merging process.
+            Merge-MaybeNullInput -InputObject $InputObject -LclPSBoundParameters $PSBoundParameters
+
+            Set-PSDebug 0
+
+            # if ((-not $PSBoundParameters.ContainsKey('WorkloadIdentityEnabled')) -and $InputObject.PSObject.Properties['WorkloadIdentityEnabled']) {
+            #     $WorkloadIdentityEnabled = $InputObject.WorkloadIdentityEnabled
```

Comment:
`WorkloadIdentityEnabled` and `OidcIssuerProfileEnabled` below will be used by code outside the inputOject parsing block, so even after calling `Merge-MaybeNullInput`, we'll still need to assign values. 

Unfortunately no easy way to get rid of the ugly parsing ;(

Reply:
Where are these used?  They are checked in PSBoundParameters but won't that be OK?

Reply:
Actually you're right, they are checked in PSBoundParameters https://github.com/xw-zhang24/azure-powershell/blob/generation/src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1#L829